module-4/hydra/README.md

```diff
@@ -5,14 +5,107 @@ Hydra can be used to TBD.
 
 This is a lightweight implementation that covers the following features/options:
 
-- ...
-- ...
-- ...
+| Tag               | description                                                 | required |
+| ----------------- | ----------------------------------------------------------- | -------- |
+| `-u` `--username` | Username for SSH login                                      | x        |
+| `-s` `--server`   | Server IP or DNS for SSH login                              | x        |
+| `-w` `--wordlist` | Wordlist for dictionary attack                              |          |
+| `--min`           | Minimum length of passwords for brute force (default: 4)    |          |
+| `--max`           | Maximum length of passwords for brute force (default: 8)    |          |
+| `-c` `--charset`  | Charset for brute force attack (default: lowercase letters) |          |
+
+## Features
+
+- **Brute Force Attack**: Tries all possible password combinations within a specified range of lengths and characters.
+- **Dictionary Attack**: Attempts to log in using passwords from a provided wordlist.
+- **Customizable Character Set and Length**: Allows for customization of the character set and password length for brute force attacks.
+- **Command-Line Interface**: Easy-to-use command-line options for setting parameters.
 
 ## Getting started
 
-TBD
+### Prerequisites
+
+- Python 3.12.3
+- `paramiko` 2.12
+- `argparse`
```

Comment:
small sentence about what this is, see previous comment for [notes](https://github.com/JohnTwiiX/dso-python-tasks/pull/1/files#r1667764240) - this should be changed and not just resolved